src/Atmos/Model/AtmosModel.jl

```diff
@@ -1142,4 +1150,250 @@ function numerical_flux_first_order!(
     end
 end
 
+function numerical_flux_first_order!(
+    numerical_flux::RoeNumericalFluxMoist,
+    balance_law::AtmosModel,
+    fluxᵀn::Vars{S},
+    normal_vector::SVector,
+    state_prognostic⁻::Vars{S},
+    state_auxiliary⁻::Vars{A},
+    state_prognostic⁺::Vars{S},
+    state_auxiliary⁺::Vars{A},
+    t,
+    direction,
+) where {S, A}
+    balance_law.moisture isa EquilMoist ||
+        error("Must use a EquilMoist model for RoeNumericalFluxMoist")
+    numerical_flux_first_order!(
+        CentralNumericalFluxFirstOrder(),
+        balance_law,
+        fluxᵀn,
+        normal_vector,
+        state_prognostic⁻,
+        state_auxiliary⁻,
+        state_prognostic⁺,
+        state_auxiliary⁺,
+        t,
+        direction,
+    )
+
+    FT = eltype(fluxᵀn)
+    param_set = balance_law.param_set
+    _cv_d::FT = cv_d(param_set)
+    _T_0::FT = T_0(param_set)
+    γ::FT = cp_d(param_set) / cv_d(param_set)
+    _e_int_v0 = e_int_v0(param_set)
+    Φ = gravitational_potential(balance_law, state_auxiliary⁻)
+
+    ρ⁻ = state_prognostic⁻.ρ
+    ρu⁻ = state_prognostic⁻.ρu
+    ρe⁻ = state_prognostic⁻.ρe
+    ρq_tot⁻ = state_prognostic⁻.moisture.ρq_tot
+
+    u⁻ = ρu⁻ / ρ⁻
+    e⁻ = ρe⁻ / ρ⁻
+    ts⁻ = recover_thermo_state(balance_law, state_prognostic⁻, state_auxiliary⁻)
+    h⁻ = total_specific_enthalpy(ts⁻, e⁻)
+    qt⁻ = ρq_tot⁻ / ρ⁻
+    c⁻ = soundspeed_air(ts⁻)
+
+    ρ⁺ = state_prognositc⁺.ρ
+    ρu⁺ = state_prognostic⁺.ρu
+    ρe⁺ = state_prognostic⁺.ρe
+    ρq_tot⁺ = state_prognostic⁺.moisture.ρq_tot
+
+    u⁺ = ρu⁺ / ρ⁺
+    e⁺ = ρe⁺ / ρ⁺
+    ts⁺ = recover_thermo_state(balance_law, state_prognostic⁺, state_auxiliary⁺)
+    h⁺ = total_specific_enthalpy(ts⁺, e⁺)
+    qt⁺ = ρq_tot⁺ / ρ⁺
+    c⁺ = soundspeed_air(ts⁺)
+    ũ = roe_average(ρ⁻, ρ⁺, u⁻, u⁺)
+    e_tot = roe_average(ρ⁻, ρ⁺, e⁻, e⁺)
+    h̃ = roe_average(ρ⁻, ρ⁺, h⁻, h⁺)
+    qt = roe_average(ρ⁻, ρ⁺, qt⁻, qt⁺)
+    ρ = sqrt(ρ⁻ * ρ⁺)
+    e_int⁻ = internal_energy(ts⁻)
+    e_int⁺ = internal_energy(ts⁺)
+    e_int = roe_average(ρ⁻, ρ⁺, e_int⁻, e_int⁺)
+    ts = PhaseEquil(
+        param_set,
+        e_int,
+        ρ,
+        qt,
+        balance_law.moisture.maxiter,
+        balance_law.moisture.tolerance,
+    )
+    c̃ = sqrt(roe_average(ρ⁻, ρ⁺, c⁻^2, c⁺^2))
+    _cv_m = cv_m(ts)
+    R_m = gas_constant_air(ts)
+    _cp_m = cp_m(ts)
+    gamma = _cp_m / _cv_m
+    # chosen by fair dice roll
+    # guaranteed to be random
+    ω = FT(π) / 3
+    δ = FT(π) / 5
+    random_unit_vector = SVector(sin(ω) * cos(δ), cos(ω) * cos(δ), sin(δ))
+    # tangent space basis
```

Comment:
Do any two tangential vectors τ1 τ2 work ? 
Do you need to add assert(  norm(τ1) > 0 ) ? I guess the random_unit_vector is to avoid random_unit_vector // normal_vector ?

Reply:
Yes pretty much, I don't think norm(τ1) = 0 is a possibility.